### ISSUE TITLE
Memory usage and pre-train without caffe weight

### ISSUE THREAD

**Issue by John1231983** [hellochick/PSPNet-tensorflow#9]
It is very impressive work. Thanks for sharing it
I am using TiTanX pascal 12GB and I cannot train in your current setting. So, I wonder how much GPU did you use when training? I need reduce the batch size to 2 but the result is so bad

Second things, I do not want to use pre-train weight from caffe. Could we use pre-train from Imagenet with resnet from official website at https://github.com/zhengyang-wang/Deeplab-v2--ResNet-101--Tensorflow? I want to know how can we obtain the result without information from PSPNet caffe.

Thanks

**Comment by hellochick** [comment]
Hey @John1231983 , I can only use batch size 2 on my 1080ti, so it's not your problem. And my result was bad too. The big problem is the author use MPI or openMP to train on multi-gpu, collecting batch info from them. I have not  found the solution yet.

I think you can use pre-train weight from imagenet with resnet, and you need the resnet model with dilatiation convolution instead of the origin. Then change the fc layers into Pyramid Pooling module. But the problem of batch size is still exist. We should find a way to update mean/var first.

**Comment by John1231983** [comment]
Great. I also know the problem of PSPnet when the author publish his code in caffe. They did not open source code of caffe with optimize gpu. I remember someone computed about 4 gpu need to train the model.

**Comment by hellochick** [comment]
Yes, the author use 4 gpu to train the model. So, if we want to reproduce the result described in the paper, we first need to deal with batch norm layer with openMP.

**Comment by shipengai** [comment]
Hi , all, @hellochick,@John1231983.
     when use small batch to train .how about the result on ade20k dataset?
     Thanks.

**Comment by shipengai** [comment]
@hellochick Thanks for your reply. I find your train.py in your project. So, is the train.py useless?
Did you train PSPnet on ADE20K dataset or  cityscapes dataset? 
I'm looking forward to your reply.
Thanks.

**Comment by shipengai** [comment]
@hellochick if you have trained PSPnet , can you share me some experience ?
 Once I wanted to train PSPnet with your code. However,I found it lacks the part of the loading pretraining model(resnet50 with the dilated network strategy),so the training result is so bad.

**Comment by hellochick** [comment]
@shipeng-uestc, the training code can finetune the model while frozen the bn parameters. You can refer to issue #11, It is hard to train PSPNet from scratch since we have limit GPU memory, unless you can implement SyncBN layer that mentioned in original paper.

**Comment by pewpewpeww** [comment]
hi, where can I change the batch size?
